### PR TITLE
Document limitations with the Google OIDC connector and transitive group memberships

### LINKED ISSUE #10635: Google Workspace OIDC connector can become unusable when fetching transitive groups if a user belongs to external groups
### Description

When using the OIDC connector for Google Workspace configured to fetch transitive groups, users that belong to external Google Groups are not able to login, as the [`groups.memberships.searchTransitiveMemberships`](https://cloud.google.com/identity/docs/reference/rest/v1/groups.memberships/searchTransitiveMemberships) API will return a somewhat generic `Unauthorized` error.

This is doubly problematic because - unless external groups are disabled at the google workspace level - anyone is seemingly able to add a user to an external group without requiring confirmation.

### Potential solutions

We could just accept this limitation (as it is, in fact, a Google Workspace limitation) and document the fact that transitive group membership listing only works if the users do not belong to external groups - and to be completely fair, we already indirectly required this by saying that the `google_admin_email` user must have view permissions over all the groups that users belong to, which is just never true if any user belongs to an external group. Something we ought to mention in the docs is that there is a way to [block users from joining external groups](https://support.google.com/a/answer/9468710) at the workspace level.

The `searchTransitiveMemberships` call can select just the ["security" groups](https://support.google.com/a/answer/10607394); this prevents the issue as those groups don't allow members outside of the Google Workspace, and thus should be always readable with group admin read privileges. This is a breaking change, however - potentially a welcome one, as it would allow admins to select specific groups to be analyzed for role mapping, but a breaking one nonetheless. Adding a configuration option increases "configuration fatigue" however, which is also not great.

We could inject a new `security_groups` claim and either duplicate the regular groups that are also security groups in there (which would bloat the user traits even more), or only list the security groups in there and not in the `groups` claim (which would be a breaking change); this would allow us to try to fetch just the security groups when fetching the full list fails for whatever reason - this seems more awkward than adding a configuration option, however.

Falling back to the old direct membership API is also an option, but will result in potentially confusing behavior if for some reason the transitive group API starts failing (and potentially insecure, if some groups are mapped to roles with `deny` rules). An error when logging in is far clearer than suddenly having access to half of the nodes that you expect for no apparent reason.

Another option could be to manually enumerate the groups; this still needs some way to select the behavior, and it's unclear how many API calls to the "Admin SDK Directory API" we can do in burst - this is the worst possible option because we need to recursively call [`groups.list`](https://developers.google.com/admin-sdk/directory/reference/rest/v1/groups/list) on the user and then _each and every returned group_, recursively.

### PR DESCRIPTION
**Pull request by espadolini**
Closes #10635, albeit unsatisfactorily.

### CODE REVIEW

**Review by ptgott — COMMENTED** [review]
It's not clear to me why this note should have the starting/ending point that it does. I think it would be clearer to break the "Configuring Google Workspace" H2 into H3-level sections. 

The first could be a "Configure Google group permissions" (or similar) section that rephrases the the six paragraphs after the "Configuring Google Workspace" heading into a set of instructions. 

Then we can change each bullet point into an H3-level heading so this organization is clearer. For example, the "Create a new project" bullet would instead be a "### Create a new project" H3 heading, then the first line in that section would be a sentence saying, "Next, [create a new project](https://console.cloud.google.com/projectselector2/apis/dashboard)."

What do you think? I'm also happy to implement this suggestions as a PR against your branch if that's easier for you.

**Comment by espadolini** [comment]
The initial part of "Configuring Google Workspace" is currently also acting as an implicit "choose which mode of operation you want" step; it would be better if we made it more explicit - especially if we also tied it into the "upgrade" instructions that are currently kind of awkwardly placed at the end of "Create an OIDC connector".

Given my mediocre technical writing skills we'd probably be better off if you implemented those changes - feel free to just push to this PR rather than opening up a new one about it, my only goal was to have *something* documented about this weird behavior that we've discovered before more customers stumble upon it inadvertently, but from what I've seen, configuring the google workspace connector is still kind of a pain point so we should probably document it better.

**Review by zmb3 — APPROVED** [review]
Looks good to me as long as Paul's comments are addressed.

File: docs/pages/enterprise/sso/google-workspace.mdx [hunk 1]
```diff
@@ -172,10 +172,11 @@ Google Admin Console, which relies on the Cloud Identity API.
 If your Google Workspace plan does not support the Cloud Identity API, you must
 ensure that your RBAC does not depend on transitive group memberships.
 
-Regardless of your plan, we recommend blocking access to external groups at the
-workspace level (following
-[these instructions](https://support.google.com/a/answer/9468710)) to prevent
-unexpected results when Teleport queries group memberships.
+When querying transitive group memberships we recommend blocking access to
```
Comment (ptgott) [inline docs/pages/enterprise/sso/google-workspace.mdx:175]:
```suggestion
When querying transitive group memberships, we recommend blocking access to
```

File: docs/pages/enterprise/sso/google-workspace.mdx [hunk 2]
```diff
@@ -172,10 +172,11 @@ Google Admin Console, which relies on the Cloud Identity API.
...
+external groups at the workspace level (following [these
+instructions](https://support.google.com/a/answer/9468710)), as membership into
```
Comment (ptgott) [inline docs/pages/enterprise/sso/google-workspace.mdx:177]:
```suggestion
instructions](https://support.google.com/a/answer/9468710)), as membership in
```

**Review by ptgott — APPROVED** [review]
Left some minor suggestions, but this works for me!

**Pull request merged**